### PR TITLE
Add scripts for visual object detection

### PR DESCRIPTION
**Pull request by caisq**
- Uses the TFHub version of SSD on MobileNetV2
  - Latency on each 1024x768 input image is acceptable: <0.2 s
    (CPU inference)
- Refactor convert_events_to_tsv_rows into its own module
  (events.py), so that it can be reused by audio_events.py
  and object_detection.py
- Add test images and video file to testdata.
- Add command line examples for the detect_objects.py binary to
  README.md. Includes example for a video file (e.g., .mp4) as the input
  and example for a series of image files (e.g., .jpg files) as the input.

### CODE REVIEW

File: Observer/SpeakFasterObserver Decoder/object_detection.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    cv2.imwrite(tmp_image_path, image)     # save frame as JPEG file
+    yield read_image(tmp_image_path), timestamp
+    timestamp += 1.0 / frame_rate
+    success, image = video_cap.read()
+  tf.io.gfile.remove(tmp_image_path)
+
+
+def get_video_fps(video_file_path):
+  """Gets the frame rate (fps) of given video file."""
+  video_cap = cv2.VideoCapture(video_file_path)
```
Comment (vsubhashini) [inline Observer/SpeakFasterObserver Decoder/object_detection.py:74]: Does this work for screen captured video format?

File: Observer/SpeakFasterObserver Decoder/object_detection_test.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    self.assertEqual(image_tensor.shape, (1, 768, 1024, 3))
+    self.assertEqual(image_tensor.dtype, tf.uint8)
+
+  def testReadImagesByGlobPattern(self):
+    generator = object_detection.read_images("testdata/*.jpg", frame_rate=4.0)
+    output = list(generator)
+    self.assertLen(output, 3)
+    for i in range(3):
+      image_tensor, _ = output[i]
+      self.assertEqual(image_tensor.shape, (1, 768, 1024, 3))
```
Comment (vsubhashini) [inline Observer/SpeakFasterObserver Decoder/object_detection_test.py:27]: Is the detector reshaping the input image appropriately?
Sorry, I'm unable to see the full detector file at once so still figuring this out.

**Review by vsubhashini — APPROVED** [review]
Minor clarification questions.

File: Observer/SpeakFasterObserver Decoder/detect_objects.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+parser.add_argument(
+    "--input_image_glob",
+    help="Glob pattern for image inputs. "
+    "Mutually exclusive with --input_video_path")
+parser.add_argument(
+    "--frame_rate",
+    default=None,
+    type=float,
+    help="Frame rate of the images from the glob pattern. "
+    "Required if --input_imgae_glob is specified.")
```
[the remark this answers is not in the document]
Comment (caisq) [reply Observer/SpeakFasterObserver Decoder/detect_objects.py]: Done.

File: Observer/SpeakFasterObserver Decoder/object_detection.py (hunk 1 above)
Comment (caisq) [reply Observer/SpeakFasterObserver Decoder/object_detection.py:74]: Screen capture can be saved in any format, including the tested .mp4 format. Once we know what format we'll be getting from the Observer, we can revisit this script about compatibility.

Comment (caisq) [reply Observer/SpeakFasterObserver Decoder/object_detection.py:74]: @TeamGleasonDev 
@joncamp

**Review by caisq — COMMENTED** [review]
Thanks for the review. Comments are addressed.

File: Observer/SpeakFasterObserver Decoder/object_detection_test.py (hunk 2 above)
Comment (caisq) [reply Observer/SpeakFasterObserver Decoder/object_detection_test.py:27]: The MobileNetV2 SSD model is able to accommodate different image sizes. The asserted 768x1024 is the actual size of the .jpg image.

**Comment by caisq** [comment]
@TeamGleasonDev Any remaining comments or questions about this CL? If not, can we merge it?

**Pull request merged**